allenact/algorithms/onpolicy_sync/runner.py

```diff
@@ -630,50 +655,77 @@ def experiment_name(self):
+        if self.save_dir_fmt == SaveDirFormat.NESTED.value:
+            folder = os.path.join(self.output_dir, *path_parts, "checkpoints",)
+        elif self.save_dir_fmt == SaveDirFormat.FLAT.value:
+            folder = os.path.join(self.output_dir, "checkpoints", *path_parts,)
         if create_if_none:
             os.makedirs(folder, exist_ok=True)
         return folder
 
-    def log_writer_path(self, start_time_str: str) -> str:
-        path = os.path.join(
-            self.output_dir,
-            "tb",
-            self.config.tag()
-            if self.extra_tag == ""
-            else os.path.join(self.config.tag(), self.extra_tag),
-            start_time_str,
-        )
-        if self.mode == TEST_MODE_STR:
-            path = os.path.join(path, "test", self.local_start_time_str)
-        return path
-
-    def metric_path(self, start_time_str: str) -> str:
-        return os.path.join(
-            self.output_dir,
-            "metrics",
-            self.config.tag()
-            if self.extra_tag == ""
-            else os.path.join(self.config.tag(), self.extra_tag),
-            start_time_str,
-        )
+    def log_writer_path(self, start_time_str: str, checkpoint_parts: List = []) -> str:
+        if self.save_dir_fmt == SaveDirFormat.NESTED.value:
+            if self.mode == TEST_MODE_STR:
+                return os.path.join(
+                    *checkpoint_parts,
+                    "test",
+                    self.config.tag(),
+                    self.local_start_time_str,
+                )
+            path = os.path.join(
+                self.output_dir,
+                self.config.tag()
+                if self.extra_tag == ""
+                else os.path.join(self.config.tag(), self.extra_tag),
+                start_time_str,
+                "train_tb",
+            )
+            return path
+        elif self.save_dir_fmt == SaveDirFormat.FLAT.value:
+            path = os.path.join(
+                self.output_dir,
+                "tb",
+                self.config.tag()
+                if self.extra_tag == ""
+                else os.path.join(self.config.tag(), self.extra_tag),
+                start_time_str,
+            )
+            if self.mode == TEST_MODE_STR:
+                path = os.path.join(path, "test", self.local_start_time_str)
+            return path
+
+    def metric_path(self, start_time_str: str, checkpoint_parts: List = []) -> str:
+        if self.save_dir_fmt == SaveDirFormat.NESTED.value:
+            return os.path.join(
+                *checkpoint_parts, "test", self.config.tag(), start_time_str,
+            )
+        elif self.save_dir_fmt == SaveDirFormat.FLAT.value:
+            return os.path.join(
+                self.output_dir,
+                "metrics",
+                self.config.tag()
+                if self.extra_tag == ""
+                else os.path.join(self.config.tag(), self.extra_tag),
+                start_time_str,
+            )
```

Comment:
Same comments about `checkpoint_parts`, `.value`, and having an `else` as above.